repair/repair.cc

```diff
@@ -1030,9 +1033,109 @@ future<> repair::shard_repair_task_impl::release_resources() noexcept {
     neighbors = {};
     dropped_tables = {};
     nodes_down = {};
+    _update_erm_callback = {};
     return make_ready_future();
 }
 
+static future<bool> ranges_intersect(const dht::token_range_vector& ranges, const dht::token_range& range) {
+    for (const auto& r : ranges) {
+        if (range.intersection(r, dht::token_comparator()).has_value()) {
+            co_return true;
+        }
+        co_await coroutine::maybe_yield();
+    }
+    co_return false;
+}
+
+static future<bool> compare_tablets(const locator::effective_replication_map& old_erm, const locator::effective_replication_map& new_erm, table_id table_id, const std::unordered_set<locator::tablet_id>& involved_tablets) {
+    auto& old_tablet_map = old_erm.get_token_metadata().tablets().get_tablet_map(table_id);
+    auto& new_tablet_map = new_erm.get_token_metadata().tablets().get_tablet_map(table_id);
+    if (old_tablet_map.tablet_count() != new_tablet_map.tablet_count()) {
```

Comment:
This assumes when tablet count changes all the tablets are changed. This is true for now, since we change as power of two. This could be changed in the future, say we have tablets t1=(1,100] t2= (100,200] and t2 is being repaired,  we split t1 into (1,50] (50,100]. Now we have 3 tablets, but it is perfectly fine to continue to repair t2.

Reply:
Right, I think it can be safely dropped and only compare the tablet itself.

Reply:
Outdated